modules/cleanmedianetBidAdapter.js

```diff
@@ -23,53 +44,50 @@ export const helper = {
       }
     }
     return BANNER;
+  },
+  getBidFloor(bid) {
+    if (!isFn(bid.getFloor)) {
+      return bid.params.bidfloor ? bid.params.bidfloor : null;
+    }
+
+    let bidFloor = bid.getFloor({
+      mediaType: '*',
+      size: '*',
+      currency: 'USD'
+    });
+
+    if (isPlainObject(bidFloor) && !isNaN(bidFloor.floor) && bidFloor.currency === 'USD') {
+      return bidFloor.floor;
+    }
+
+    return null;
   }
 };
 
 export const spec = {
   code: 'cleanmedianet',
   aliases: [],
-  supportedMediaTypes: [BANNER, VIDEO],
+  supportedMediaTypes: ['banner', 'video'],
 
   isBidRequestValid: function (bid) {
-    return (
-      !!bid.params.supplyPartnerId &&
-      typeof bid.params.supplyPartnerId === 'string' &&
-      (typeof bid.params.bidfloor === 'undefined' ||
-        typeof bid.params.bidfloor === 'number') &&
-      (typeof bid.params['adpos'] === 'undefined' ||
-        typeof bid.params['adpos'] === 'number') &&
-      (typeof bid.params['protocols'] === 'undefined' ||
-        Array.isArray(bid.params['protocols'])) &&
-      (typeof bid.params.instl === 'undefined' ||
-        bid.params.instl === 0 ||
-        bid.params.instl === 1)
-    );
+    return !!bid.params.supplyPartnerId && isStr(bid.params.supplyPartnerId) &&
+      (!bid.params['rtbEndpoint'] || isStr(bid.params['rtbEndpoint'])) &&
+      (!bid.params.bidfloor || isNumber(bid.params.bidfloor)) &&
+      (!bid.params['adpos'] || isNumber(bid.params['adpos'])) &&
+      (!bid.params['protocols'] || Array.isArray(bid.params['protocols'])) &&
+      (!bid.params.instl || bid.params.instl === 0 || bid.params.instl === 1);
   },
 
   buildRequests: function (validBidRequests, bidderRequest) {
     return validBidRequests.map(bidRequest => {
-      const {
-        adUnitCode,
-        auctionId,
-        mediaTypes,
-        params,
-        sizes,
-        transactionId
-      } = bidRequest;
-      const baseEndpoint = 'https://bidder.cleanmediaads.com';
-      const rtbEndpoint =
-        `${baseEndpoint}/r/${
-          params.supplyPartnerId
-        }/bidr?rformat=open_rtb&reqformat=rtb_json&bidder=prebid` +
-        (params.query ? '&' + params.query : '');
-      let url = bidderRequest.refererInfo.page;
-
+      const {adUnitCode, auctionId, mediaTypes, params, sizes, transactionId} = bidRequest;
+      const baseEndpoint = params['rtbEndpoint'] || ENDPOINTS['cleanmedianet'];
```

Comment:
Can you please enforce the value from `params[rtbEndpoint]` is using `https:` protocol.